### DISCUSSION TITLE
Incorrect rendering of rivers (with pmtiles)

### DISCUSSION THREAD

**Discussion by MightyPork** [maplibre/maplibre-gl-js#4764] [Q&A] [answered]
I followed the steps on the protomaps website, got my map with `pmtiles extract`.

The map file looks fine when viewed in [an online tool](https://pmtiles.io). However, on my page, water rendering is completely bugged out. I removed all external styles and JS to isolate the problem.

**maplibre-gl-js version**: 

```
<script src="[https://unpkg.com/maplibre-gl@^4.7.1/dist/maplibre-gl.js](https://unpkg.com/maplibre-gl@%5E4.7.1/dist/maplibre-gl.js)"></script>
<link href="[https://unpkg.com/maplibre-gl@^4.7.1/dist/maplibre-gl.css](https://unpkg.com/maplibre-gl@%5E4.7.1/dist/maplibre-gl.css)" rel="stylesheet" />
<script src="https://unpkg.com/pmtiles@3.2.0/dist/pmtiles.js"></script>
```

**browser**:

tested in chrome, vivaldi, firefox

### Steps to Trigger Behavior

see the demo. The most basic usage.

### Link to Demonstration

Here is a live demo:

https://bits.ondrovo.com/osm/

### Expected Behavior

water rendering normally

### Actual Behavior

Looks as if the SVG paths were closed and filled randomly.

Rivers are increasingly broken the more you zoom in.

Buildings and roads render fine. 

![Screenshot_20240925_133034](https://github.com/user-attachments/assets/90bff22a-bbae-4c06-94c6-b8bad44ee80d)

zoomed out, rivers look more correct

![Screenshot_20240925_133050](https://github.com/user-attachments/assets/bd627ab9-e9c9-4d7f-9cd7-4a2450369b3e)

**Comment by MightyPork** [comment]
maybe I should instead report this in the pmtiles plugin github?

**Comment by HarelM** [comment, marked as answer]
I'm not aware of these kind of issues in maplibre.
Without a minimal reproduction there's not much I can do.
Most likely, the problem is related to how you created the pmtiles...
Did you try out pmtiles from protomaps examples first to see that you get good results?

**Reply by MightyPork** [reply]
Good idea, thanks for the tip.

the issue does not happen with the example file on the protomaps github -  https://pmtiles.io/protomaps(vector)ODbL_firenze.pmtiles

I exactly followed their online guide. So either the pmtiles tool or the builds at https://maps.protomaps.com/builds/ are currently broken... or not compatible with the javascript library :thinking:

**Comment by adjourn** [comment]
This looks like a styling issue, not data. I think I've had the similar issue before, can't remember specifics.

Likely you're using some kind of Maplibre layer type (e.g `fill`) with source data type (e.g `MultiLineString`) that produces the unexpected outcome. My guess based on the "glitch" is that you're styling water (or water name) linestrings with `fill`.

**Reply by acalcutt** [reply]
somewhat similar to what we saw in https://github.com/maplibre/maplibre-native/issues/1843 , were tilemaker was adding a line around the tiles, causing a weird fill issue in native. In that PR I ended up using [vt2geojson](https://github.com/mapbox/vt2geojson) to try and [pull out some specific](https://github.com/maplibre/maplibre-native/issues/1843#issuecomment-1799388230) tiles having issues, which allowed me to see the issue easier.

Edit: Though in this case, I guess it was bad data

**Comment by MightyPork** [comment]
The issue was solved by updating the styles, pmtiles was migrating the planet files hosted on their server to a new format and the style shown in examples was outdated. This new format added SVG paths to rivers, which previously only used closed polygons
